ml-agents-envs/mlagents/envs/environment.py

```diff
@@ -527,8 +541,43 @@ def _get_state(self, output: UnityRLOutputProto) -> AllBrainInfo:
             _data[brain_name] = BrainInfo.from_agent_proto(
                 self.worker_id, agent_info_list, self.brains[brain_name]
             )
+        self._parse_side_channel_message(output.side_channel)
         return _data
 
+    def _parse_side_channel_message(self, data: bytearray) -> None:
+        offset = 0
+        while offset < len(data):
+            try:
+                channel_type = struct.unpack("i", data[offset : offset + 4])[0]
+                offset = offset + 4
+                message_len = struct.unpack("i", data[offset : offset + 4])[0]
+                offset = offset + 4
+                message_data = data[offset : offset + message_len]
```

Comment:
Can you be a bit more aggressive on the bounds checks here? If `offset` or `offset + message_len` are out of range, the results are silently truncated.

I guess `if len(message_data) != message_len: raise ...` is sufficient.